propagation_test.go

```diff
@@ -38,6 +38,11 @@ func TestSpanPropagator(t *testing.T) {
 	metricsFactory, metrics := initMetrics()
 	tracer, closer := NewTracer("x", NewConstSampler(true), reporter, TracerOptions.Metrics(metrics), TracerOptions.ZipkinSharedRPCSpan(true))
 
+	// Note: the closing of the main tracer object Close also and
+	// related containers as reporter, sampler, logger & etc.
+	// So need to close it after related containers no longer used
+	defer closer.Close()
```

Comment:
not following this. The Close() was already called below. Calling it inline, rather than via defer, is technically a cleaner way to guarantee that all spans are "flushed" (even though for in-memory reporter it does't make much difference since it has no background flushing).